src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -405,9 +405,9 @@ private bool IsTagRecoveryStopPoint(HtmlSymbol sym)
 
         private void TagContent()
         {
-            if (!At(HtmlSymbolType.WhiteSpace))
+            if (!At(HtmlSymbolType.WhiteSpace) && !At(HtmlSymbolType.NewLine))
```

Comment:
Nice. Do we have other cases where a newline would confuse the parser? For example does `EndTagBlock()` in this class handle the following?
``` html
   <p
    >
```